src/main/java/com/algorand/algosdk/util/abi/values/ValueUfixed.java

```diff
@@ -0,0 +1,25 @@
+package com.algorand.algosdk.util.abi.values;
+
+import com.algorand.algosdk.util.Encoder;
+import com.algorand.algosdk.util.abi.types.TypeUfixed;
+
+import java.math.BigInteger;
+
+public class ValueUfixed extends Value {
+    public ValueUfixed(TypeUfixed ufixedT, BigInteger value) {
+        BigInteger upperLimit = BigInteger.ONE.shiftLeft(ufixedT.bitSize);
+        if (value.compareTo(upperLimit) >= 0)
+            throw new IllegalArgumentException("cannot construct ABI ufixed value: passed in value larger than ufixed size");
+        this.abiType = ufixedT;
+        this.value = value;
+    }
+
+    public ValueUfixed(int size, int precision, BigInteger value) {
+        this(new TypeUfixed(size, precision), value);
+    }
+
+    public byte[] encode() {
+        int ufixedSize = ((TypeUfixed) this.abiType).bitSize;
+        return Encoder.encodeUintToBytes(((BigInteger) this.value), ufixedSize);
```

Comment:
Should this be something like: 
```
int ufixedPrecision = ((TypeUfixed) this.abiType).precision;
return Encoder.encodeUintToBytes(((BigInteger) this.value * Math.pow(10, ufixedPrecision)), ufixedSize);
```

Reply:
actually we are working over integer rather than rational, so we just encode the numerator over power of ten 😅